server/datastore.js

```diff
@@ -254,6 +257,31 @@ const whichReferencesToDisplay = (dataPerSample, threshold=5, maxNum=10) => {
     return refMatchesAcrossSamples;
 };
 
+function split_mutations(mstring){
+    let arr = mstring.split("&");
+    return arr;
+}
+
+function getSampleVariants(){
+    let fileToParse = global.config.run.annotatedPath+"results/mutations.json";
+    let variantData ={};
+    if (!fs.existsSync(fileToParse)) {
+        //warn(`Could not find mutations file, ${fileToParse}, doesn't exist - skipping.`);
+        return variantData;
+    } 
+    let file_content =fs.readFileSync(fileToParse).toString();
+    let regex = /\,(?!\s*?[\{\[\"\'\w])/g;
```

Comment:
can we have examples of what is the file_content format and how do we want to transform it? The regex is kind of cryptic

Reply:
it's a json, for example:
`[{"barcode":"none","mutations":[]},{"barcode":"BC24","mutations":[]},{"barcode":"BC72","mutations":[]},{"barcode":"BC63","mutations":[]},{"barcode":"BC30","mutations":[]},{"barcode":"BC25","mutations":[]},
....
,{"barcode":"BC27","mutations":[{"name": "UK","mutations":[{"position":25782,"from":"C","to":"G"},{"position":13034,"from":"G","to":"A"},
....
{"position":28281,"from":"A","to":"G"},],"subs":[{"name": "UK-1","mutations":[{"position":13993,"from":"G","to":"A"},{"position":16953,"from":"A","to":"G"},{"position":29613,"from":"G","to":"A"},{"position":13729,"from":"G","to":"A"},{"position":28882,"from":"G","to":"A"},{"position":25782,"from":"C","to":"G"},{"position":13034,"from":"G","to":"A"},{"position":17814,"from":"G","to":"C"},{"position":18897,"from":"T","to":"G"},{"position":25790,"from":"G","to":"A"},{"position":11584,"from":"T","to":"C"},{"position":13666,"from":"A","to":"G"},],"subs":[]},]},]},]},{"barcode":"BC28","mutations":[]}
`
and we want to be able to parse it using JSON.parse() method.

Reply:
If it is JSON, why do we need regexp?

Reply:
in the file we are reading, there are commas before brackets. javascript cant deal with them, so we are removing them before we are parsing the file.what the regex does is simply removing that commas we dont want to be there

Reply:
but yes, that is kind of cryptic, we might consider creating a valid JSON in the first place. the json file is being created in `pipelines/SARS-CoV-2-pipelines/compare_mutations.py` in `find_variants2` function... as I am looking now, that python script needs some clean-up anyway.

Reply:
done. we don't need the regexp anymore.